src/iamserver/publicmessagehandler.cpp

```diff
@@ -110,15 +110,11 @@ bool PublicMessageHandler::IsMainNode() const
     };
 
     auto it = std::find_if(mNodeInfo.mAttrs.begin(), mNodeInfo.mAttrs.end(), [&](const auto& attr) {
-        return caseInsensitiveEqual(std::string_view(attr.mName.CStr(), attr.mName.Size()), cNodeTypeTag);
+        return caseInsensitiveEqual(std::string_view(attr.mName.CStr(), attr.mName.Size()), aos::iam::cAttrMainNode);
     });
```

Comment:
I think this logic would better to do on the config side.
Otherwise it will repeat in different components

Reply:
Could it be method of NodeInfo struct?